src/databricks/labs/blueprint/paths.py

```diff
@@ -598,46 +845,17 @@ def expanduser(self):
             return self.with_segments(homedir, *self._path_parts[1:])
         return self
 
-    def is_notebook(self):
+    def is_notebook(self) -> bool:
         """Return True if the path points to a notebook in Databricks Workspace."""
         try:
             return self._object_info.object_type == ObjectType.NOTEBOOK
         except DatabricksError:
             return False
 
-    def iterdir(self):
+    def iterdir(self) -> Generator[WorkspacePath, None, None]:
```

Comment:
```suggestion
    def iterdir(self) -> Iterable[WorkspacePath]:
```
maybe?..

Reply:
Mypy is pretty adamant that the signature needs to be like this.

I'm also mildly surprised by the this, but think it's because the superclass implementations are generator functions. Any ideas?